File: src/encoders/discussion-send-action.ts
import { DiscussionSendAction } from "~/models";

export const encodeDiscussionSendAction = (action: DiscussionSendAction, includeParentsAndStudents: boolean): DiscussionSendAction => {
  if (
    action === DiscussionSendAction.SendEveryoneExceptParentsAndStudents &&
    includeParentsAndStudents
  ) action = DiscussionSendAction.SendEveryone;

  if (
    action === DiscussionSendAction.ReplyEveryoneExceptParentsAndStudents &&
    includeParentsAndStudents
  ) action = DiscussionSendAction.ReplyEveryone;

  return action;
};

File: library/src/commonMain/kotlin/ink/literate/pawnote/encoders/discussion-send-action.kt
package ink.literate.pawnote.encoders

import ink.literate.pawnote.models.DiscussionSendAction

fun encodeDiscussionSendAction (action: DiscussionSendAction, includeParentsAndStudents: Boolean): DiscussionSendAction {
    var actionToSend: DiscussionSendAction = action

    if (
        action == DiscussionSendAction.SendEveryoneExceptParentsAndStudents &&
        includeParentsAndStudents
    ) actionToSend = DiscussionSendAction.SendEveryone

    if (
        action == DiscussionSendAction.ReplyEveryoneExceptParentsAndStudents &&
        includeParentsAndStudents
    ) actionToSend = DiscussionSendAction.ReplyEveryone

    return actionToSend
}